src/main/fc/fc_core.c

```diff
@@ -254,6 +255,16 @@ void updateArmingStatus(void)
             }
         }
 
+#ifdef USE_GPS_RESCUE
+        if (IS_RC_MODE_ACTIVE(BOXGPSRESCUE)) {
```

Comment:
Not sure if this check should be dependent on the switch position being 'GPS rescue on', this will lead to the craft going into rescue mode immediately on arming. This should probably rather be `isModeActivationConditionPresent(BOXGPSRESCUE)` (i.e. a switch and range are configured for `BOXGPSRESCUE`).

Reply:
It wouldn't go into rescue mode because one of the sanity checks is min_distance (currently fixed at 100 meters). Not even sure if this should even be a configuration setting, if you want to mess with that distance you might as well disable sanity checks.

Reply:
Ok. Even then, checking if there is a GPS fix and then disabling / enabling arming based on this **only if** the 'rescue mode' switch is on does not make much sense:
- if the pilot does not activate the switch before arming, no check will be performed;
- if the pilot activates and then deactivates the switch before there is a fix, arming will never be enabled, since the condition isn't updated.

Reply:
You're right, will fix.